frontend/src/views/producer/producer-modal/index.js

```diff
@@ -0,0 +1,103 @@
+import React, { Component } from 'react';
+import Modal, { Content } from '../../../components/modal';
+import Typography from '../../../components/typography';
+import { connect } from '../../../utils';
+import { withTheme } from 'styled-components';
+import Icon from 'antd/lib/icon';
+
+export default withTheme(
+  connect('store')(
+    class AppModal extends Component {
+      state = {
+        show: false,
+        handledSignalToShow: false,
+      };
+      constructor(props) {
+        super(props);
+        if (props.store.ui.auth.validateTokenFailed) {
```

Comment:
yes, its a very strict modal from mobx-state-tree and is guaranteed to not be null. because if its null then the state management system will throw exception first

Reply:
this component is not reusable so i dont see any reason why it cant be coupled with the structure of the state.
It would be sooo weird to couple a generic button for example, to 'buttonDisabled' in somewhere global (either React context, Redux store, or mobxStateTree', but this component is supposed to reflect precisely just that piece of state from the store. Writing getters to map will just create more obstacle and confuse people.

Reply:
Good points. 👍 

My only remaining concern is the possible future refactoring of the structure of the store. If that changes, it has to be reflected down to these components, which might be a lot of work. But I'm not sure what the alternative would be, so I think you can decide what's best here :)

Reply:
This component is still coupled with the state so I will need to change it anyway. Either this or somewrapper or function that serve to select the states

Reply:
Yes, maybe create a property for the component like this:
`auth = () => this.props.store.ui.auth;`

Then instead of repeating `this.props.store.ui.auth` everywhere you could just say `this.auth` instead and in case of the store structure is changed there would only be one place where to change it.

Reply:
class instance is not update with props though. the only place to do it is in componentWillUpdate, componentDidUpdate, and render